sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -434,10 +439,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
 
           logInfo(s"Listing $qualified on driver")
           Try(fs.listStatus(qualified)).getOrElse(Array.empty)
-        }.filterNot { status =>
-          val name = status.getPath.getName
-          name.toLowerCase == "_temporary" || name.startsWith(".")
-        }
+        }.filterNot(fileStatusFilter)
```

Comment:
I mean this disables caching `FileStatus`es of part-files, and this does influence `ParquetInputFormat.listStatus()`. I believe the reason why no test case was broken is that currently `ParquetInputFormat` only calls `listStatus()` for reading footers. But this is completely an implementation detail within Parquet itself and shouldn't be viewed as a stable assumption.